users/views/bind_company_view.py

```diff
@@ -0,0 +1,277 @@
+
+    Allows authenticated users to associate themselves with either:
+    1. An existing company (startup or investor)
+    2. A newly created company if no matching company exists
+
+    POST data format:
+    {
+        "company_name": "Company Name",
+        "company_type": "startup"  # or "investor"
+    }
+    """
+    permission_classes = [IsAuthenticated]
+
+    def post(self, request):
+        """
+        Handle company binding request.
+
+        Args:
+            request: HTTP request containing company_name and company_type
+
+        Returns:
+            Response with binding result or error details
+        """
+        serializer = CompanyBindingSerializer(data=request.data)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        company_name = serializer.validated_data['company_name']
+        company_type = serializer.validated_data['company_type']
+        user = request.user
+
+        if self._is_user_bound_to_company(user):
+            return Response(
+                {"error": "User is already bound to a company."},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        try:
+            with transaction.atomic():
+                if company_type == 'startup':
+                    return self._bind_to_startup(user, company_name)
+                else:
+                    return self._bind_to_investor(user, company_name)
+
+        except ValidationError as e:
+            return Response({"error": str(e)}, status=status.HTTP_400_BAD_REQUEST)
+        except Exception as e:
+            logger.error(f"Error binding company: {str(e)}")
+            return Response(
+                {"error": "An unexpected error occurred."},
+                status=status.HTTP_500_INTERNAL_SERVER_ERROR
+            )
+
+    def _is_user_bound_to_company(self, user):
+        """
+        Check if user is already associated with any company.
+
+        Args:
+            user: User instance to check
+
+        Returns:
+            bool: True if user is bound to a company, False otherwise
+        """
+        return hasattr(user, 'startup') or hasattr(user, 'investor')
+
+    def _bind_to_startup(self, user, company_name):
+        """
+        Bind user to an existing or new startup.
+
+        Args:
+            user: User instance to bind
+            company_name: Name of the startup
+
+        Returns:
+            Response with binding result
+        """
+        try:
+            startup = Startup.objects.get(company_name__iexact=company_name)
+
+            if hasattr(startup, 'user') and startup.user is not None:
+                return Response(
+                    {"error": "Startup is already associated with another user."},
+                    status=status.HTTP_400_BAD_REQUEST
+                )
+
+            startup.user = user
+            startup.save()
```

Comment:
No transaction rollback if startup fails to save, risking partial updates. Wrap assignment and save in try-except and raise for rollback.
```
try:
    startup.user = user
    startup.save()
except Exception as e:
    raise ValidationError(str(e))
```